lib/chef/licensing.rb

```diff
@@ -0,0 +1,94 @@
+require "chef-licensing"
+require_relative "log"
+require_relative "licensing_config"
+
+class Chef
+  class Licensing
+    class << self
+      def fetch_and_persist
+        Chef::Log.info "Fetching and persisting license..."
+        license_keys = ChefLicensing.fetch_and_persist
+      rescue ChefLicensing::LicenseKeyFetcher::LicenseKeyNotFetchedError
+        Chef::Log.error "Chef Infra cannot execute without valid licenses." # TODO: Replace Infra with the product name dynamically
+        Chef::Application.exit! "License not set", 174 # 174 is the exit code for LICENSE_NOT_SET defined in lib/chef/application/exit_code.rb
+      rescue ChefLicensing::SoftwareNotEntitled
+        Chef::Log.error "License is not entitled to use Chef Infra."
+        Chef::Application.exit! "License not entitled", 173 # 173 is the exit code for LICENSE_NOT_ENTITLED defined in lib/chef/application/exit_code.rb
+      rescue ChefLicensing::Error => e
+        Chef::Log.error e.message
+        Chef::Application.exit! "Usage error", 1 # Generic failure
+      end
+
+      def check_software_entitlement!
+        Chef::Log.info "Checking software entitlement..."
+        ChefLicensing.check_software_entitlement!
+      rescue ChefLicensing::SoftwareNotEntitled
+        Chef::Log.error "License is not entitled to use Chef Infra."
+        Chef::Application.exit! "License not entitled", 173 # 173 is the exit code for LICENSE_NOT_ENTITLED defined in lib/chef/application/exit_code.rb
+      rescue ChefLicensing::Error => e
+        Chef::Log.error e.message
+        Chef::Application.exit! "Usage error", 1 # Generic failure
```

Comment:
same

Reply:
This is not needed in this PR and removed it.